wallet/transaction.js

```diff
@@ -160,21 +160,21 @@ class WalletAddress {
   }
 }
 
-class WalletSecretString {
-  constructor(secretString) {
-    this.secretString = secretString
+class HdNode {
+  constructor(hdNodeString) {
+    this.hdNodeString = hdNodeString
   }
 
   getSecretKey() {
-    return this.secretString.substr(0, 128)
+    return this.hdNodeString.substr(0, 128)
```

Comment:
Is there any specific reason not to do these in the constructor?

Reply:
not really, we can rewrite it. The original idea was to keep the "hdNodeString" as the only source of truth and to derive all the remaining information, but in hindsight I agree that it would probably be cleaner to split the "hdNodeString" in the constructor and we can save those getters